packages/design-system/src/components/Drawer/Drawer.stories.jsx

```diff
@@ -60,17 +60,43 @@ const Template = ({ data, ...args }) => (
       fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
       officia deserunt mollit anim id est laborum.
     </p>
-  </Drawer>
+  </>
 );
 
+const Template = ({ data, ...args }) => <Drawer {...args}>{drawerContent}</Drawer>;
+
 export const DrawerDefault = Template.bind({});
 export const DrawerWithStickyPositioning = Template.bind({});
 DrawerWithStickyPositioning.args = {
   isFooterSticky: true,
   isHeaderSticky: true,
 };
-export const DrawerToggleDefault = () => (
-  <DrawerToggle drawerOpen={false} showDrawer={() => {}}>
-    Drawer Toggle{' '}
-  </DrawerToggle>
+
+function WithState({ children }) {
+  const [state, setState] = useState({});
+  return <div>{children(state, setState)}</div>;
+}
+export const DrawerToggleWithDrawer = () => (
+  <WithState>
+    {(state, setState) => (
```

Comment:
Love that this uses hooks now. Is the `WithState` wrapper function necessary?

Reply:
Could it just use state inside `DrawerToggleWithDrawer`?

Reply:
Fyi I'm working through the same kind of issue right now with pagination but using this: https://medium.com/urbandataanalytics/changing-args-values-state-in-storybook-without-usestate-hook-15cc096716f7

args instead of state

Reply:
works great! here is my snippet that works if it helps:

```
const Template = ({...args}) => {
  const [{currentPage}, updateArgs] = useArgs();
  const handleSetPage = (evt, page) => {
    evt.preventDefault();
    updateArgs({ currentPage: page })
  }
  return (
    <>
      <div className="ds-u-text-align--center">
        Current page: {currentPage} / {args.totalPages}
      </div>
      <Pagination
        {...args}
        onPageChange={handleSetPage}
        renderHref={(page) => `#/results/${page}`}
      />
    </>
  );
}```

Reply:
Updated using args for state management!